k-distribution/src/main/scripts/lib/py-kframework/pyk.py

```diff
@@ -0,0 +1,240 @@
+#!/usr/bin/env python3
+
+import sys
+import subprocess
+
+from kast import *
+
+def match(pattern, kast):
```

Comment:
what is this being used for? it doesn't seem to deal with sorts very intelligently.

Reply:
No, it certainly doesn't. It's not meant to be a robust pattern matcher.

It's used for doing simple rewrites/manip over the AST before handing it over to K. Often it's easier to specify the desired behavior as a rewrite (which uses the matcher) than to manually traverse/update the term.

This is used in function `normalizeKLabels`, for example, which does things like replacing `true and TERM` with just `TERM`.